openedx/core/djangoapps/course_apps/api.py

```diff
@@ -42,4 +56,9 @@ def set_course_app_enabled(course_key: CourseKey, app_id: str, enabled: bool, us
     """
     course_app = CourseAppsPluginManager.get_plugin(app_id)
     enabled = course_app.set_enabled(course_key, user=user, enabled=enabled)
+    CourseAppStatus.objects.update_or_create(
```

Comment:
Once _fetch_ status is decoupled `update_status_for_course_app` This can use the same method to update app status.